File: scripts/Convert a Number to Hexadecimal/Convert a Number to Hexadecimal.cpp
class Solution
{
public:
    string toHex(int num)
    {
        string hex = "0123456789abcdef";
        unsigned int n = num; // to handle neg numbers
        string ans = "";
        if (n == 0)
            return "0";
        while (n > 0)
        {
            int k = n % 16;
            ans += hex[k];
            n /= 16;
        }
        reverse(ans.begin(), ans.end()); // as we stored it in the opposite order
        return ans;
    }
};

File: algorithms/C/Convert a Number to Hexadecimal/Convert a Number to Hexadecimal.py
class Solution:
	def toHex(self, num: int) -> str:
		ret = ["0","1","2","3","4","5","6","7","8","9","a","b","c","d","e","f"]
		ans = ""

		if num < 0:
			num = pow(2,32) +num

		if num == 0:
			return "0"
		while num > 0:
			ans = ret[num%16] +ans
			num = num//16

		return ans
